### PR TITLE
create vsphere dev branch for different model

### PR DESCRIPTION
**Pull request by msfuko**
Hi @hidetosaito,

Please kindly help review the PR, it's for the model changes of vsphere service. Thank you very much.

Hi @AnnaRoelle and @falau,

Please kindly review the models for 
* esxi - https://github.com/msfuko/qa-sample-service/tree/vsphere-dev/dcsqa/model/esxi
* vm - https://github.com/msfuko/qa-sample-service/tree/vsphere-dev/dcsqa/model/vm
Thank you very much.

Regards,
Chloe

### CODE REVIEW

File: dcsqa/criteria.py [hunk 1]
```diff
@@ -79,29 +73,34 @@ def set_criteria_by_ticketkey_host():
         current_app.logger.warn("couldn't parse JSON %s" % ex)
         return response.bad_request("invalid JSON format")
 
+    if data.get('isVirtual'):
+        from dcsqa.model.vm.criteria import CriteriaData
+    else:
+        from dcsqa.model.esxi.criteria import CriteriaData
+
```
Comment (hidetosaito) [inline dcsqa/criteria.py]: :+1: 
very cool :sunglasses:

**Comment by hidetosaito** [comment]
HI @msfuko 

it works well on my side, very good quality :+1: 
i have some questions, 

1. this implementation contains only for ```/vsphere/v1/{criteria, raw, result} ``` services, is that correct? 
2. how about vm ? are you willing to define ```KIND='vm'``` ? or switched by ```isVirtual``` with same URI?
3. i think Reference Implementation here is good enough already for new developer, so i recommend to push to https://github.com/TrendMicroDCS/qa-vsphere-worker (with some branches if needs)  to reduce your maintain effort

**Comment by msfuko** [comment]
Hi @hidetosaito,

Thanks for the review :-P
For the questions I'd like to consult from you about some topic as well.

1. yes, you're correct
2. host and vm are both using vsphere sdk, currently we didn't separate to be so fine-grained. originally make server type as the path is also a choice. e.g. /vsphere/v1/host/{criteria, raw, result} and /vsphere/v1/vm/{criteria, raw, result}. However, using the same URI format to all the services might be much consistent for current phase. Please kindly advise if making type into URI is more reasonable.
3. we only remain 18/20 private repo for current annual plan, and gitlab system seems like be postponed.  maybe we could use different branches for now :blush: developers can also have their own corresponding remote branches and push to it accordingly. The only concern is that currently it's public repo. Please kindly advise if we need to make it as private and use different branches.

Thank you very much for your guidance. 

Regards,
Chloe

**Comment by hidetosaito** [comment]
Hi @msfuko 

For vm part, i'll send my thought via e-mail to you later.
For private repo, i have been cleaned it up now, so feel free to create private repo as needed.

**Pull request merged**